### PR TITLE
feat: ensures all demo pages have unique/descriptive document titles

### LINKED ISSUE #32: update document.title on demo pages
a static title of `Deque Cauldron React` is an issue. Prepend the default with descriptive info of the current page (`Radiogroup | Component demo | Deque Cauldron React` for example)

### PR DESCRIPTION
**Pull request by schne324**
Closes #32

### CODE REVIEW

File: demo/Demo/index.js [hunk 1]
```diff
@@ -18,6 +19,9 @@ class Demo extends Component {
 
     return (
       <div className="Demo">
+        <Helmet>
+          <title>{`${displayName} | Component demo`}</title>
```
Comment (schne324) [inline demo/Demo/index.js]: react-helmet adds an unwanted `,` in the title if its not all one string...

When I did `<title>{displayName} | Component demo</title>`, it outputted `Button, | Component demo...` as the title. See https://github.com/nfl/react-helmet/issues/408#issuecomment-499763409

**Comment by schne324** [comment]
This doesn't completely cover #32 because not all components use the `<Demo />` component yet. The plan is to move every demo over eventually but this is a good start

**Comment by stephenmathieson** [comment]
I think @scurker's suggestion is great and am _almost_ sold on the idea, but I don't want to have to manually add "title" information when declaring a route. I'd much rather have it magically done for me.

**Comment by scurker** [comment]
Assuming title can be the link text that we're already passing in there shouldn't be that much additional work.

https://github.com/dequelabs/cauldron-react/blob/58d73041f5292429b8409a6b94796f2ce020762b/demo/index.js#L154-L156

**Comment by schne324** [comment]
> I don't want to have to manually add "title" information

☝️ completely agree, which is why I thought the demo component would be perfect. However, as @scurker pointed out we can simply rely on the existing link text which basically means when you add a new component and setup the routing/sidebar link, then you get the document title for free.

All in favor of going with @scurker 's suggestion?

**Comment by stephenmathieson** [comment]
Works for me.

**Comment by schne324** [comment]
Meh, I'm still on the fence now that I think about it...relying on the text only relieves us of half of the annoyance.

we'd still have to _remember_ to do this every time we add a route
```
    <Route component={({ location }) => {
          <ReactHelmet title={location.state.title} />
        }} />
```

With the demo component we basically never have to think about it

**Comment by scurker** [comment]
> Meh, I'm still on the fence now that I think about it...relying on the text only relieves us of half of the annoyance.
> 
> we'd still have to _remember_ to do this every time we add a route

Why would you have to remember to do it every time you add route? `<Route component={...} />` should match _any_ route so as long as the location state contains a title (set via `<Link>`) there should be nothing else to add.

**Comment by schne324** [comment]
@scurker oh duh...sorry I misread that suggestion. Great call 👍

**Pull request merged**